spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -26,6 +26,21 @@
     end
   end
 
+  describe 'bad omniauth session' do
+    let(:omniauth_response_nil_uid) do
+      OmniAuth::AuthHash.new(provider: 'barcode', uid: nil, info: { last_name: "" })
+    end
+    before do
+      controller.request.env['omniauth.auth'] = omniauth_response_nil_uid
+    end
+    it 'handles nil UID' do
+      allow(User).to receive(:from_barcode) { valid_barcode_user }
```

Comment:
If the message is '...could not authenticate because barcode or last name did not match active patron..', can we use an invalid_barcode_user in the test?

Reply:
The user itself is irrelevant for this test, what causes the error is the `nil` UID in the OmniAuth session.